### PR TITLE
Feature/cvsb 2867

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: src/functions/retroGen.ts [hunk 1]
```diff
@@ -16,24 +17,23 @@ const retroGen: Handler = async (event: any, context?: Context, callback?: Callb
...
-                return s3BucketService.upload(`cvs-retro-reports-${process.env.BUCKET}`, generationServiceResponse.fileName, generationServiceResponse.fileBuffer)
-                    .then((result: any) => {
-                        return result;
-                    });
-            })
-            .catch((error: any) => {
-                console.log(error);
-            });
+        .then(async (generationServiceResponse: { fileName: string, fileBuffer: Buffer }) => {
+            const tokenResponse = await sharepointAuthenticationService.getToken();
```
Comment (KHodderDVSA) [inline src/functions/retroGen.ts]: Need to be careful with error handling on remote services when using async/await - it will always return something, but it will ignore errors. Need to handle remote service failures somehow.

File: src/services/SharePointAuthenticationService.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            uri: "https://login.microsoftonline.com/6c448d90-4ca1-4caf-ab59-0a2aa67d7801/oauth2/token",
+            form: {
+                grant_type: "client_credentials",
+                client_id: this.spConfig.azure_sharepoint_client_id,
+                client_secret: this.spConfig.azure_sharepoint_client_secret,
+                resource: "https://graph.microsoft.com/"
+            }
+        };
+
+        return rp(tokenParams);
```
Comment (KHodderDVSA) [inline src/services/SharePointAuthenticationService.ts]: Can do unit tests on this with sinon or proxyquire (for mocking anonymous default exports - see test-number branch 6136 for examples of this). just enough to keep Sonar happy.

File: src/utils/Configuration.ts [hunk 3]
```diff
@@ -52,15 +55,12 @@ class Configuration {
      * Retrieves the DynamoDB config
      * @returns IDynamoDBConfig
      */
-    public getS3Config(): IS3Config {
-        if (!this.config.s3) {
-            throw new Error("DynamoDB config is not defined in the config file.");
+    public getSharePointConfig(): ISPConfig {
```
Comment (KHodderDVSA) [inline src/utils/Configuration.ts]: can also test configuration. See test-results 5909 branch PR

File: sonar-project.properties [hunk 4]
```diff
@@ -9,7 +9,7 @@ sonar.projectVersion=1.0
 # This property is optional if sonar.modules is set.
 sonar.sources=src
 
-sonar.exclusions=.build, src/models/**, src/config/**, src/services/LambdaService.ts, src/services/S3BucketService.ts, src/handler.ts
+sonar.exclusions=.build, src/models/**, src/config/**, src/services/LambdaService.ts, src/handler.ts, src/services/SharePointAuthenticationService.ts, src/services/SharePointService.ts
```
Comment (KHodderDVSA) [inline sonar-project.properties]: Not a fan of just excluding stuff for the sake of it. Good that they're so simple as to not need hardly any testing, but should add what little we can.

File: tests/unit/reportGen.unitTest.ts [hunk 5]
```diff
@@ -197,35 +197,4 @@ describe("report-gen", () => {
             });
         });
     });
-
```
Comment (KHodderDVSA) [inline tests/unit/reportGen.unitTest.ts]: why was this removed? Should it not be somewhere?

File: src/services/SharePointAuthenticationService.ts (hunk 2 above)
Comment (oalexandru) [reply src/services/SharePointAuthenticationService.ts]: Didn't manage to mock the imports using any mocking library. Came up with an alternative that it's a little bit hacky.

File: src/utils/Configuration.ts (hunk 3 above)
Comment (oalexandru) [reply src/utils/Configuration.ts]: done

File: sonar-project.properties (hunk 4 above)
Comment (oalexandru) [reply sonar-project.properties]: Removed the services from exclusions
